dcc-import-gene/src/main/java/org/icgc/dcc/imports/gene/writer/GeneWriter.java

```diff
@@ -17,139 +17,478 @@
+            } else if (entry.get("type").asText().equals("start_codon")) {
+              curTranscript.put("start_exon", exons.size() - 1);
+            } else if (entry.get("type").asText().equals("stop_codon")) {
+              curTranscript.put("end_exon", exons.size() - 1);
+            }
+
+          }
+        }
+      }
+    }
   }
 
   @Override
-  public void writeFiles(@NonNull InputStream inputStream) {
-    write(inputStream, all());
+  public void writeFiles(@NonNull ObjectNode value) {
+    if (++counter % STATUS_GENE_COUNT == 0) {
+      log.info("Writing {}", counter);
+    }
+    this.geneCollection.insert(value);
   }
 
-  @SneakyThrows
-  public void write(@NonNull InputStream geneStream, @NonNull GeneFilter geneFilter) {
-    val watch = createStarted();
-    log.info("Writing gene model into {}...", mongoUri);
+  /**
+   * Responsible for parsing the lines of the gtf file and producing generic ObjectNodes
+   * @param String represnting the current line in the gtf file
+   * @return ObjectNode representation of the gtf row.
+   */
+  private ObjectNode parseLine(@NonNull String s) {
+    String[] line = TSV.split(s);
+    val seqname = line[0].trim();
+    val source = line[1].trim();
+    val type = line[2].trim();
+    String locStart = line[3].trim();
+    String locEnd = line[4].trim();
+
+    Double score;
+    try {
+      score = Double.parseDouble(line[5].trim());
+    } catch (Exception var15) {
+      score = 0.0D;
+    }
+
+    char strand = line[6].trim().charAt(0);
+    int locationStart = Integer.parseInt(locStart);
+    int locationEnd = Integer.parseInt(locEnd);
+    if (locationStart > locationEnd) {
+      int location = locationStart;
+      locationStart = locationEnd;
+      locationEnd = location;
+    }
+
+    val negative = locationStart <= 0 && locationEnd <= 0;
+    assert strand == 45 == negative;
+
+    int frame;
+    try {
+      frame = Integer.parseInt(line[7].trim());
+    } catch (Exception var14) {
+      frame = -1;
+    }
+
+    val attributes = line[8];
+    val attributeMap = parseAttributes(attributes);
+
+    ObjectNode feature = MAPPER.createObjectNode();
 
-    // Open BSON file stream
-    val genes = readGenes(geneStream);
+    int strandNumber = 0;
+    if (strand == '+') {
+      strandNumber = 1;
+    } else if (strand == '-') {
+      strandNumber = -1;
+    }
+
+    feature.put("seqname", seqname);
+    feature.put("source", source);
+    feature.put("type", type);
+    feature.put("locationStart", locationStart);
+    feature.put("locationEnd", locationEnd);
+    feature.put("score", score);
+    feature.put("strand", strandNumber);
+    feature.put("frame", frame);
 
-    writeGenes(watch, genes, geneFilter);
+    for (val kv : attributeMap.entrySet()) {
+      feature.put(kv.getKey(), kv.getValue());
+    }
+
+    return feature;
+  }
+
+  private String getName(String symbol) {
+    if (nameMap.containsKey(symbol)) {
+      return nameMap.get(symbol);
```

Comment:
use `get` with default. Using contains is always a smell for this type of access.